sympy/calculus/util.py

```diff
@@ -709,6 +709,131 @@ def is_convex(f, *syms, **kwargs):
         return False
     return True
 
+
+def stationary_points(f, symbol, domain=S.Reals):
+    """
+    Returns the stationary points of a function (where derivative of the
+    function is 0) in the given domain.
+
+    Parameters
+    ==========
+
+    f : Expr
+        The concerned function.
+    symbol : Symbol
+        The variable for which the stationary points are to be determined.
+    domain : Interval
+        The domain over which the stationary points have to be checked.
+        If unspecified, S.Reals will be the default domain.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, S, sin, log, pi, pprint, stationary_points
+    >>> from sympy.sets import Interval
+    >>> x = Symbol('x')
+
+    >>> stationary_points(1/x, x, S.Reals)
+    EmptySet()
+
+    >>> pprint(stationary_points(sin(x), x), use_unicode=False)
+              pi                              3*pi
+    {2*n*pi + -- | n in Integers} U {2*n*pi + ---- | n in Integers}
+              2                                2
+
+    >>> stationary_points(sin(x),x, Interval(0, 4*pi))
+    {pi/2, 3*pi/2, 5*pi/2, 7*pi/2}
+
+    """
+    from sympy import solveset, diff
+
+    if isinstance(domain, EmptySet):
+        return S.EmptySet
+
+    domain = continuous_domain(f, symbol, domain)
+    set = solveset(diff(f, symbol), symbol, domain)
+
+    return set
+
+
+def maximum(f, symbol, domain=S.Reals):
```

Comment:
`maximum` name is potentially overloaded. More specific names are better.

Reply:
Originally, I used `maximize` for this. Because of https://github.com/sympy/sympy/blob/4063057bbe136d1a2a1d4a6b325565259eadeb84/sympy/utilities/tests/test_wester.py#L2248-L2258
I changed the name according to @asmeurer's suggestion (https://github.com/sympy/sympy/pull/16473#issuecomment-478083631).
Should I change it back?

Reply:
Note that here, `test_U14` is still failing because `function_range` is unable to find range in this case. But `test_U13` is passing after this change

Reply:
It can be discussed. My concern is that `maximum` is too generic a name. But it might be OK.